tools/helper/src/client/utils/getHeaders.ts

```diff
@@ -0,0 +1,143 @@
+export interface Header {
+  /**
+   * The level of the header
+   *
+   * `1` to `6` for `<h1>` to `<h6>`
+   */
+  level: number
+  /**
+   * The title of the header
+   */
+  title: string
+  /**
+   * The slug of the header
+   *
+   * Typically the `id` attr of the header anchor
+   */
+  slug: string
+  /**
+   * Link of the header
+   *
+   * Typically using `#${slug}` as the anchor hash
+   */
+  link: string
+  /**
+   * The children of the header
+   */
+  children: Header[]
+}
+
+export type HeaderLevels = false | number | [number, number] | 'deep'
+
+// cached list of anchor elements from resolveHeaders
+const resolvedHeaders: { element: HTMLHeadElement; link: string }[] = []
+
+export type MenuItem = Omit<Header, 'slug' | 'children'> & {
+  element: HTMLHeadElement
+  children?: MenuItem[]
+}
+
+export interface GetHeadersOptions {
+  /**
+   * The selector of the headers
+   *
+   * @default '#vp-content :where(h1,h2,h3,h4,h5,h6)'
+   */
+  selector?: string
+  /**
+   * Ignore specific elements within the header.
+   *
+   * @default []
+   */
+  ignore?: string[]
+  /**
+   * The levels of the headers
+   *
+   * - `false`: No headers.
+   * - `number`: only headings of that level will be displayed.
+   * - `[number, number]: the first number is the minimum level and the second number is the maximum level.
+   * - `deep`: same as `[2, 6]`, which means all headings from `<h2>` to `<h6>` will be displayed.
+   *
+   * @default 2
+   */
+  levels?: HeaderLevels
+}
+
+export function getHeaders({
+  selector = '#vp-content :where(h1,h2,h3,h4,h5,h6)',
```

Comment:
I still want this be ahigh support range selector,  Maybe creating with  Array.map

Reply:
I don't think that overly flexible parameter types are necessary. As a selector, strings are already sufficient to meet the requirements, without assuming more possibilities.

Using `string[]` with `ignore` is because the nodes to be ignored may be isolated from each other without any inherent correlation.

Reply:
I was trying to say that we can avoid using where by creating a long selector string from Array.map

Reply:
As a default setting, I believe as long as it complies with the 'guideline', it should suffice. Through my practical testing, the compatibility of `:where` is actually better than what is stated in [caniuse](https://caniuse.com/mdn-css_selectors_where). Despite being marked as 'support unknown' in some mobile browsers, it is actually supported in browsers like `UC` and `quark`. As a utility function, the default setting for parameters should not be overdesigned. The final decision on how to pass them in should be left to the function caller, to be handled according to their specific scenario.